spec/features/bookmarks_spec.rb

```diff
@@ -8,7 +8,7 @@
   end
   scenario "user can see boomarks index" do 
     visit topic_path(@topic)
-    expect(page).to have_css("h4", text: @topic.bookmarks.first.url)
+    expect(page).to have_xpath("//img[contains(@src, @bookmark.url)]")
```

Comment:
Try to stick with CSS selectors rather than xpath selectors as web developers on the whole are more familiar with CSS:
```ruby
expect(page).to have_css("img[src='#{@bookmark.url}']"
```